test/06-extra-args/CMakeLists.txt

```diff
@@ -7,7 +7,66 @@ find_package(Pytest REQUIRED)
 enable_testing()
 
 pytest_discover_tests(
-    TestExtraArgs
+    TestExtraArgs.extraArgs
     EXTRA_ARGS "--capture=no" "--cmdopt=demo"
     DEPENDS test_extra_args.py
 )
+
+set(EXPECTED
+    "TestExtraArgs.extraArgs.test_requires_no_capture"
+    "TestExtraArgs.extraArgs.test_requires_args"
+)
+
+add_test(
+    NAME TestExtraArgs.Validate
+    COMMAND ${CMAKE_COMMAND}
+    -D "TEST_PREFIX=TestExtraArgs.extraArgs"
+    -D "EXPECTED=${EXPECTED}"
+    -P ${CMAKE_CURRENT_SOURCE_DIR}/../01-modify-name/compare_discovered_tests.cmake
+)
+
+# Negative Test which suppresses test discovery
+pytest_discover_tests(
+    TestDiscoveryArgsSupressDiscovery.discoverExtraArgs
+    DISCOVERY_EXTRA_ARGS "-v"
+    DEPENDS test_extra_args.py
+)
```

Comment:
Using `-v` or `-q` as an extra argument for discovery is discouraged, so I wouldn't recommend using it as an example for unit tests. I think it might be best to choose a real use case, such as `-k EXPRESSION` or `--ignore=dir/`.